### PR TITLE
ci: Include tracking of GOOS for which Caddy fails to build

### PR DESCRIPTION
**Pull request by Mohammed90**
We learned about the build failure for Solaris by accident. Perhaps it's better we have track of which OSs Caddy fails to build for to have an understanding our areas of improvements and where we are probably unable to support.

Updates #3615

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Used as inspiration: https://github.com/mvdan/github-actions-golang
 
-name: Cross-Platform
+name: Cross-Platform Tests
```
Comment (francislavoie) [inline .github/workflows/ci.yml:3]: Idk I think we can shorten this frankly

```suggestion
name: Tests
```

File: .github/workflows/cross-build.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Cross-Build (Advisory Check)
```
Comment (francislavoie) [inline .github/workflows/cross-build.yml:1]: :man_shrugging: 

```suggestion
name: Cross-Build
```

**Comment by francislavoie** [comment]
FYI @mholt unfortunately github actions doesn't have a "warning" state for jobs yet, but there's an open issue discussing it here https://github.com/actions/toolkit/issues/399

So for now, we have `continue-on-error: true` set for this stuff so that it doesn't make the PR go red for any cross-build failures, but the only other option is green. To see the failures you can look at the annotations on the workflow: https://github.com/caddyserver/caddy/actions/runs/187524216

**Comment by mholt** [comment]
This looks cool, neat idea!

I can see why this would be even better with some sort of "warning" result instead of just green/red.

I guess something we can add to our TODO list is to submit patches upstream, or at least request that they make their build tags more correct, so that we can continue to build x-platform.

**Comment by Mohammed90** [comment]
> Is this ready to merge or should we wait? Is it even useful if it always emits a green status?

It is ready. The results always being green bothers me too, but I hope the workflow being listed reminds to check for annotations... until Github Actions supports tri-state results.

**Comment by Mohammed90** [comment]
Blocked by #3625 😓 (kinda, sorta)

**Comment by mholt** [comment]
My bad. Will get that figured out soon.

**Comment by mholt** [comment]
Okie all fixed

**Pull request merged**